daemon/logger/logger.go

```diff
@@ -16,10 +20,52 @@ type Message struct {
 	Timestamp   time.Time
 }
 
-// Logger is interface for docker logging drivers
+// Logger is the interface for docker logging drivers
 type Logger interface {
 	Log(*Message) error
 	Name() string
 	Close() error
-	GetReader() (io.Reader, error)
+}
+
+type ReadConfig struct {
+	Since  time.Time
+	Tail   int
+	Follow bool
+}
+
+// LogReader is the interface for reading log messages for loggers that support reading
+type LogReader interface {
+	// Read logs from underlying logging backend
+	ReadLogs(ReadConfig) *LogWatcher
+}
+
+// LogWatcher is used when consuming logs read from the LogReader interface
+type LogWatcher struct {
+	// For sending log messages to a reader
+	Msg chan Message
+	// For sending error messages that occur while while reading logs
+	Err           chan error
+	closeNotifier chan struct{}
+}
+
+// NewLogWatcher returns a new LogWatcher.
+func NewLogWatcher() *LogWatcher {
+	return &LogWatcher{
+		Msg:           make(chan Message, 10),
+		Err:           make(chan error, 10),
+		closeNotifier: make(chan struct{}, 1),
+	}
+}
+
+func (w *LogWatcher) Close() error {
+	select {
+	case w.closeNotifier <- struct{}{}:
```

Comment:
I don't understand this code. Is it possible to call Close() twice? I think you should just close channel and be done with it.